prov/efa/src/efa_mr.c

```diff
@@ -60,39 +63,39 @@ static struct fi_ops efa_mr_cache_ops = {
 int efa_mr_cache_entry_reg(struct ofi_mr_cache *cache,
 			   struct ofi_mr_entry *entry)
 {
-	int fi_ibv_access = IBV_ACCESS_LOCAL_WRITE;
-
-	struct efa_mr *md = (struct efa_mr *)entry->data;
-
-	md->domain = container_of(cache->domain, struct efa_domain,
-				  util_domain);
-	md->mr_fid.fid.ops = &efa_mr_cache_ops;
-
-	md->mr_fid.fid.fclass = FI_CLASS_MR;
-	md->mr_fid.fid.context = NULL;
-
-	md->mr = ibv_reg_mr(md->domain->ibv_pd, entry->info.iov.iov_base,
-			    entry->info.iov.iov_len, fi_ibv_access);
-	if (!md->mr) {
-		EFA_WARN_ERRNO(FI_LOG_MR, "ibv_reg_mr", errno);
-		return -errno;
-	}
-
-	md->mr_fid.mem_desc = (void *)(uintptr_t)md->mr->lkey;
-	md->mr_fid.key = md->mr->rkey;
-
-	return 0;
+	int ret = 0;
+	/* TODO
+	 * By default fi_ibv_access is set to FI_SEND | FI_RECV |
+	 * FI_REMOTE_READ | FI_REMOTE_WRITE. Once access field is
+	 * available when cache.add_region is invoked,
+	 * this can be removed
```

Comment:
This comment is vague. It should state that the current MR cache implementation does not support passing access ,  and because of this shortcoming, we had no choice but to pass all possible access when allocating new MR. We will revisit it once MR cache can support access.

Reply:
Updated, Thanks